model/src/pyrenew/arrayutils.py

```diff
@@ -92,3 +94,158 @@ def pad_x_to_match_y(
+        The sampled quantity.
+    """
+
+    value: ArrayLike | None = None
+
+    def __repr__(self):
+        return f"PeriodicProcessSample(value={self.value})"
+
+
+class PeriodicBroadcaster:
+    r"""
+    Broadcast arrays periodically using either repeat or tile,
+    considering period size and starting point.
+    """
+
+    def __init__(
+        self,
+        data_starts: int,
+        period_size: int,
+        broadcast_type: str,
+    ) -> None:
+        """
+        Default constructor for PeriodicBroadcaster class.
+
+        Parameters
+        ----------
+        data_starts : int
+            Relative point at which data starts, must be between 0 and
+            period_size - 1.
+        period_size : int
+            Size of the period.
+        broadcast_type : str
+            Type of broadcasting to use, either "repeat" or "tile".
+
+        Notes
+        -----
+        See the sample method for more information on the broadcasting types.
+
+        Returns
+        -------
+        None
+        """
+
+        self.validate(
+            data_starts=data_starts,
+            period_size=period_size,
+            broadcast_type=broadcast_type,
+        )
+
+        self.period_size = period_size
+        self.data_starts = data_starts
+        self.broadcast_type = broadcast_type
+
+        return None
+
+    @staticmethod
+    def validate(
+        data_starts: int, period_size: int, broadcast_type: str
+    ) -> None:
+        """
+        Validate the input parameters.
+
+        Parameters
+        ----------
+        data_starts : int
+            Relative point at which data starts, must be between 0 and
+            period_size - 1.
+        period_size : int
+            Size of the period.
+        broadcast_type : str
+            Type of broadcasting to use, either "repeat" or "tile".
+
+        Returns
+        -------
+        None
+        """
+
+        # Period size should be a positive integer
+        assert isinstance(
+            period_size, int
+        ), f"period_size should be an integer. It is {type(period_size)}."
+
+        assert (
+            period_size > 0
+        ), f"period_size should be a positive integer. It is {period_size}."
+
+        # Data starts should be a positive integer
+        assert isinstance(
+            data_starts, int
+        ), f"data_starts should be an integer. It is {type(data_starts)}."
+
+        assert (
+            0 <= data_starts
+        ), f"data_starts should be a positive integer. It is {data_starts}."
+
+        assert data_starts <= period_size - 1, (
+            "data_starts should be less than or equal to period_size - 1."
+            f"It is {data_starts}. It should be less than or equal "
+            f"to {period_size - 1}."
+        )
+
+        # Broadcast type should be either "repeat" or "tile"
+        assert broadcast_type in ["repeat", "tile"], (
+            "broadcast_type should be either 'repeat' or 'tile'."
```

Comment:
```suggestion
            "broadcast_type should be either 'repeat' or 'tile'. "
```

spacing so it looks right when printed on a single line